config/routes.rb

```diff
@@ -93,13 +93,16 @@
         end
       end
     end
-    resources :teams, except: [:edit, :update, :destroy] do
+    resources :teamsets, only: [:index, :edit, :update] do
       member do
-        patch :dissolve
-      end
-      collection do
-        patch :dissolve_all
+        patch :clone
         patch :randomize
+        patch :dissolve_all
+      end
+      resources :teams, only: [:show] do
```

Comment:
On one hand, yes. And it would shorten URLs for assignments (who needs a course number?), submissions (who needs the assignment, either?), graders (who needs a submission?), etc. On the other I simply can't wrap my head around those shorter URIs: it feels really deeply brittle to say "I have this one entity number, and everything else will always be perfectly intact and there will never be any data integrity issues, so we can trust this one number." I don't buy it, I've already caught several logic bugs by having these nested URLs and cross checking the various IDs against each other.